Process multiple trx files

ryepup opened the pull request:
* Treat each CLI argument as a new TRX file
* update docs
* bump version

ryepup committed:
Process multiple trx files

* Treat each CLI argument as a new TRX file
* update docs
* bump version

ryepup commented:
refs #14

gfoidl commented on source/trx2junit/Program.cs:
```diff
@@ -39,22 +40,24 @@ private static async Task RunAsync(string[] args)
             DateTime start = DateTime.Now;
 
-            var utf8 = new UTF8Encoding(false);
+            await Task.WhenAll(args.Select(Convert));
```
Does this work with `trx2junit results/*.trx` as allowed in the description?
I.e. is there every file in the path transformed.

Maybe there should be a test covering the scenarios for the argumentes given.
If you won't add one, I'll add the tests after this PR is done.

ryepup replied:
it works in bash, because the wildcard gets expanded by the shell before calling the program. I think it'd be non-trivial to add wildcard support in C#, maybe just update the docs to specify that the example only works on shells that support wildcards?

ryepup replied:
Updated docs to be clearer about the shell handling the wildcards.

I think https://www.nuget.org/packages/Microsoft.Extensions.FileSystemGlobbing could be a solution for handling older shells, but my needs are met by this implementation so I'm unlikely to expand in that direction.

gfoidl replied:
I'm with you. That looks good. If there is demand for older shells and anyone raises an issue, this can be revisited. But now it's just fine.

gfoidl replied:
For your info: https://github.com/gfoidl/trx2junit/pull/17 adds some tests for this.

gfoidl commented on source/trx2junit/Program.cs:
```diff
@@ -39,22 +40,24 @@ private static async Task RunAsync(string[] args)
+        private static async Task Convert(string trxFile){
+            string jUnitFile = Path.ChangeExtension(trxFile, "xml");
+            Console.WriteLine($"Converting '{trxFile}' to '{jUnitFile}'");
+            var utf8 = new UTF8Encoding(false);
```
The encoding is for every file the same, so we could just reference a (static readonly) field instead of creating an instance for every file.

ryepup replied:
updated

gfoidl reviewed (changes requested):
Please check the concern about folder-arguments.
Plus one small change.

ryepup committed:
Remove repetitive computation

We don't need to make a new `Encoding` every loop, all conversions can
share the same UTF8 instance

ryepup committed:
Update docs

Specify wildcard constraint. Some shells do not expand.

Future: maybe depend on `Microsoft.Extensions.FileSystemGlobbing` to
cover shells that don't support wildcards?
